featurebyte/api/target.py

```diff
@@ -108,6 +114,21 @@ def horizon(self) -> Optional[str]:
         except RecordRetrievalException:
             return self.internal_horizon
 
+    @property  # type: ignore
+    @substitute_docstring(
+        DEFINITION_DOCSTRING.format(
+            object_type="feature",
+            example=textwrap.dedent(
+                """
+                >>> target = catalog.get_target("InvoiceCount_60days")  # doctest: +SKIP
+                >>> target_definition = target.definition  # doctest: +SKIP
+                """
+            ).strip(),
+        )
+    )
+    def definition(self) -> str:  # pylint: disable=missing-function-docstring
+        return self._generate_definition()
```

Comment:
@YS-L @jevonyeoh 

Would like to get the early feedback of this pattern.

Reply:
Ya, thinking of making the decorator as a class too to make it handle those indentation issue. If we follow this pattern, it should helps to reduce the code lines of these api objects. Personally, i think it is more readable (focus more on the code & the differences).

Reply:
I think I would also lean towards repeating docstrings for simplicity. Though, I do like that in your approach, we can have more consistent docstrings. 🤔 

Separately, some other tools might not pick up programatically generated docs. For example, in the IDE itself, mouseover docs won't have the full docstring.

![Screenshot 2023-07-11 at 4 12 16 PM](https://github.com/featurebyte/featurebyte/assets/2313101/54e9e4eb-fd1f-406f-9881-ce5940a120d0)

Reply:
Ya, seems like pycharm does not show the docstring. Tested jupyter notebook just now, it can render the docstring properly.

Reply:
The main reason of doing this is because i found there are some discrepancies between similar methods in `Feature` and `Target` (`primary_entity` in feature vs `entities` in target). If we start doing it now, it will force us to revise only single docstring. In addition, we may also need to check the consistencies between two classes.

Reply:
@jevonyeoh @YS-L 

The decorator is revised here to make it clearer (like having different sections string).